### PR TITLE
fix(parser): panics and wrong readings from what a server sent back

### PR DESCRIPTION
**Pull request by lollipopkit**
From the full-tree audit. Seven findings in `sbm_parser`, which the app runs **in-process over FFI** — a panic here is the app disappearing, not an exception a caller can catch.

Two test files, both of which fail without the source changes:
- `tests/hostile_input.rs` — what a remote host must not be able to do to the process.
- `tests/field_output.rs` — output real machines produce that was read wrongly or not at all.

## Crashes

| Input | Old behaviour |
|---|---|
| `{"DeviceID":"C:","Size":512,...}` | `attempt to divide by zero` at `windows.rs:122` — `size == 0` was guarded, but anything under a KiB floors to 0 KiB and was divided by. Integer division by zero panics in release too, so this was not gated on debug assertions. |
| `df` size column `10G°` | `end byte index 4 is not a char boundary` — the unit was taken by splitting one *byte* from the end. Split by character now. |
| `"NumberOfLogicalProcessors":100000000000` | `for i in 0..logical` allocating a `CpuCore` with a heap id each, until the process is killed. `sysctl -n hw.ncpu` is the same shape on BSD. |

Both panic messages are reproduced verbatim by the tests against the pre-fix source. The unbounded allocation is bounded by `MAX_CPU_CORES` (4096, well above real hardware — the largest shipping x86 parts are in the hundreds of threads); past it the reading is a broken query and is handled the way a broken `LoadPercentage` already was, by skipping the entry. A second, valid processor entry is still read, and its cores are still numbered as though the broken one contributed nothing.

## Readings that parsed and were wrong

- **`/proc/stat` columns are `user nice system idle iowait irq softirq`.** Field 2 was read as `sys` and field 3 as `nice`. `lib/view/page/server/detail/view.dart:623` renders `ss.cpu.sys` under the label `sys` — that number was nice time, near zero on most machines, and real system time was displayed nowhere. `total()` sums every field, so every aggregate agreed with itself and no test could notice. **Nothing locked the old answer**: no Rust or Dart test asserted either field, and `nice` is rendered nowhere.
- **A wrapped `df` row lost its numbers.** `df` puts an overlong source on a line of its own; the buffered name was written *over* field 0 rather than inserted in front of it, which destroyed the size column and left five fields for a check that wants six. A stock Ubuntu LVM install (`/dev/mapper/ubuntu--vg-ubuntu--lv`) therefore reported no root filesystem at all. The finding also claimed the buffer is never cleared — it is, at the end of every non-wrapped row, and a test covers that the name applies to the next row only.
- **FreeBSD reported no network at all.** Its `netstat -ibn` has an `Idrop` column between `Ierrs` and `Ibytes`, so rows are 12 wide where macOS's are 11, and only 11 was accepted. The byte columns are now counted from the right, which reads both widths — and also a row whose interface has no link-layer address, which is one narrower again.
- **Thermal names and values were paired across two independent `cat` globs.** A zone whose `temp` cannot be read drops a line from one side only, shifting every later reading onto the preceding zone's name. Mismatched counts now report no temperatures, because a plausible wrong label is worse than a gap. Pairing at the source is a command-manifest change and would mean reissuing every installed status script, so it is noted in the function's doc rather than done here.

## Compatibility

`tests/dart_compat.rs` (92) and `tests/script_compat.rs` pass unchanged — the macOS 11-column netstat case, the aligned thermal case and the existing `df` fixtures all still assert what they did. The only deliberate divergence from the Dart implementation is the `/proc/stat` column order, which the Dart side had wrong too.

**Release note**: the `sys` figure on the server detail page changes. It was nice time; it is now system time, and will read higher on most machines.

## Tests

`cargo test --workspace` clean, `cargo clippy -p sbm_parser --all-targets` clean, and `flutter test test/frb_parser_test.dart` against a rebuilt `sbm_ffi` (17 passed).

## Summary by CodeRabbit

- **Bug Fixes**
  - Corrected Linux CPU usage reporting and improved handling of wrapped disk-statistic entries.
  - Improved temperature readings when sensor data is incomplete or includes blank lines.
  - Improved BSD network statistics across macOS and FreeBSD formats.
  - Added safeguards for invalid or unusually large CPU counts on BSD and Windows.
  - Prevented disk-size edge cases from causing calculation errors or crashes.
  - Improved handling of multibyte human-readable disk sizes without parsing failures.

- **Tests**
  - Added coverage for malformed, incomplete, and platform-specific system statistics.

### CODE REVIEW

**Comment by lollipopkit** [comment]
@coderabbitai review

File: crates/sbm_parser/src/windows.rs [hunk 1]
```diff
@@ -42,9 +42,23 @@ pub fn parse_cpu(raw: &str, prev: &[CpuCore]) -> Vec<CpuCore> {
...
+        // that is the app dying, not an exception a caller can catch. A count
+        // past the bound is a broken query, handled like a broken
+        // LoadPercentage: skip the entry rather than invent a reading for it.
+        let logical = processor["NumberOfLogicalProcessors"]
+            .as_u64()
+            .unwrap_or(physical);
+        if logical > MAX_CPU_CORES {
+            continue;
+        }
+        let logical = logical as usize;
```
[the remark this answers is not in the document]
Comment (lollipopkit) [reply crates/sbm_parser/src/windows.rs:68]: Valid — fixed in cc571d41.

The bound was on one entry, and entries each well inside it still add up: 100 records of 128 logical processors is not a machine, and every core allocated carries a heap id. The running total is what the bound applies to now. An entry that would pass it stops the loop rather than being skipped, because no later entry can bring the total back down; what was read before it is kept.

Per-entry validation and the conversion are unchanged, and a zero-logical entry stays valid and contributes nothing — `a_zero_logical_entry_is_not_a_broken_one` covers that alongside `processor_entries_cannot_add_up_past_the_bound`.

File: crates/sbm_parser/src/linux.rs [hunk 2]
```diff
@@ -300,14 +315,23 @@ pub fn parse_net(raw: &str) -> Vec<NetIface> {
...
+/// is a change to the command manifest, which every installed status script
+/// would have to be reissued for.
 pub fn parse_temps(types_raw: &str, values_raw: &str, divisor: f64) -> Temperatures {
     let mut temps = Temperatures::default();
-    let types: Vec<&str> = types_raw.split('\n').collect();
-    let values: Vec<&str> = values_raw.split('\n').collect();
+    let types: Vec<&str> = types_raw.lines().filter(|l| !l.trim().is_empty()).collect();
+    let values: Vec<&str> = values_raw.lines().filter(|l| !l.trim().is_empty()).collect();
+    if types.len() != values.len() {
+        return temps;
```
[the remark this answers is not in the document]
Comment (lollipopkit) [reply crates/sbm_parser/src/linux.rs:332]: Right about the cause, but not doing it here.

The premise checks out: `script.rs:781` is `exec 2>/dev/null`, so a `cat` that cannot read a zone's `temp` contributes no line at all, and the two streams go out of step with nothing in the output saying so. Placeholders at the source are the only thing that can recover those readings, and this parser cannot reconstruct them.

Scope is why it is not in this PR. The commands are embedded in the status script installed on each server, so changing them means a new script emitted, the `script_compat.rs` and `dart_compat.rs` fixtures updated, the monitor's extended cycle following, and — as you say — the parser accepting both formats until every server has re-installed. That is a change to the collection contract, not parser hardening, and it wants its own PR and its own e2e run against a real host.

What this PR does is make the failure safe: mismatched counts report no temperatures instead of shifting every later reading onto the preceding zone's name. A gap is recoverable on the next poll; a plausible wrong label is not. The follow-up is named in `parse_temps`'s doc comment so it is not lost.
